### PR TITLE
[Discipline] Add Amirdrassil 4p Smite to places & Misc Fixes

### PR DESCRIPTION
**Pull request by Saeldur**
### Description
- Fix a bunch of outdated ability cooldowns/categorisations
- Remove some outdated spells from showing in the list
- Add Amirdrassil Shadow Smite to all smite talents
- Fix Ultimate Penance Cooldown (It was referring to the damage spell not the cast spell)
- Fix Heavens Wrath (Outdated implementation)

### Testing
- https://www.warcraftlogs.com/reports/Np7YBKTDxz9qCkn4#fight=3&type=summary&source=87 - Blaze / TE Fixes
- https://www.warcraftlogs.com/reports/xBQfWD2Mg8A1qdzh#fight=81&type=healing&source=13 - Ultimate Penitence Fixes

### CODE REVIEW

**Comment by Krealle** [comment]
LGTM

File: src/analysis/retail/priest/discipline/modules/spells/BlazeOfLight/BlazeOfLight.tsx [hunk 1]
```diff
@@ -34,7 +39,7 @@ class BlazeOfLight extends Analyzer {
     this.addEventListener(
       Events.damage
         .by(SELECTED_PLAYER)
-        .spell([SPELLS.SMITE, SPELLS.PENANCE, SPELLS.DARK_REPRIMAND_DAMAGE]),
+        .spell([SPELLS.SMITE, SPELLS.SHADOW_SMITE, SPELLS.PENANCE, SPELLS.DARK_REPRIMAND_DAMAGE]),
```
Comment (trevorm4) [inline src/analysis/retail/priest/discipline/modules/spells/BlazeOfLight/BlazeOfLight.tsx]:
```suggestion
        .spell(BLAZE_OF_LIGHT_SPELLS),
```

File: src/analysis/retail/priest/discipline/modules/spells/TwilightEquilibrium/TwilightEquilibrium.tsx [hunk 2]
```diff
@@ -100,12 +100,24 @@ class TwilightEquilibrium extends Analyzer {
...
   checkDamageEvent(event: DamageEvent) {
+    // Shadow Smite is buffed by Holy TE's Whitelist and has no associated cast.
+    if (event.ability.guid === SPELLS.SHADOW_SMITE.id) {
+      if (
+        this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id, event.timestamp)
+      ) {
+        return true;
+      }
+
+      return false;
```
Comment (trevorm4) [inline src/analysis/retail/priest/discipline/modules/spells/TwilightEquilibrium/TwilightEquilibrium.tsx]:
```suggestion
      return this.selectedCombatant.hasBuff(SPELLS.TWILIGHT_EQUILIBRIUM_HOLY_BUFF.id, event.timestamp);
```

**Review by trevorm4 — CHANGES_REQUESTED** [review]
small nits and then lgtm

**Review by trevorm4 — CHANGES_REQUESTED** [review]
small nits and then lgtm

**Comment by emallson** [comment]
looks like everything is addressed, gonna merge shortly

**Pull request merged**